packages/tables/src/Concerns/HasColumns.php

```diff
@@ -134,27 +129,14 @@ public function setColumnValue(string $column, string $record, $input): ?array
             ];
         }
 
-        if ($columnRelationship = $column->getRelationship($record)) {
-            $record = $columnRelationship->getResults();
-            $columnName = $column->getRelationshipTitleColumnName();
-        } elseif (
-            $this instanceof HasRelationshipTable &&
-            (($tableRelationship = $this->getRelationship()) instanceof BelongsToMany) &&
-            in_array($columnName, $tableRelationship->getPivotColumns())
-        ) {
-            $record = $record->{$tableRelationship->getPivotAccessor()};
-        } else {
-            $columnName = (string) Str::of($columnName)->replace('.', '->');
-        }
-
-        if (! ($record instanceof Model)) {
-            return null;
-        }
-
-        $record->setAttribute($columnName, $input);
-        $record->save();
+        $state = $column->evaluate($column->getSaveStateUsing(), [
```

Comment:
I think that instead of doing this work in the HasColumns trait, we can just add a `$column->saveState($state)` method to the CanSaveState trait. Then we don't need the `getSaveStateUsing()` getter at all as we have access to `$this->saveStateUsing`.